src/sdp_data/database/postgresql.py

```diff
@@ -0,0 +1,101 @@
+from typing import List, Any
+
+import pandas as pd
+import psycopg2
+from psycopg2 import sql
+
+from exceptions import ColumnNamesError, NumberColumnsError
+
+
+def connect_to_database(account_type: str):
+    """
+    Connects to the PostgreSQL database by reading a file in which the accesses are stored.
+
+    The read file has the following structure:
+
+    │  account_type      │     user      │  password    │ database_name │ host │ port │
+    │  read-access       │ johndoe       │              │               │      │      │
+    │  read-write-access │ janedoe       │              │               │      │      │
+
+    To have access of the database, ask the team.
+
+    :param account_type: either 'read-access' or 'read-write-access'
+    :return: the database connector
+    """
+    filepath = "../database_access.csv"
+    database_access = pd.read_csv(filepath, header=0, index_col="account_type")
+
+    return psycopg2.connect(
+        database=database_access.loc[account_type, "database_name"],
+        user=database_access.loc[account_type, "user"],
+        password=database_access.loc[account_type, "password"],
+        host=database_access.loc[account_type, "host"],
+        port=database_access.loc[account_type, "port"],
+    )
+
+
+def truncate_and_insert(db_connector: Any, df: pd.DataFrame, schema: str, table_name: str):
+    """
+    Overwrites the destination table with the source data contained in the dataframe.
+    It is done by truncating (or emptying) the destination table and then inserting values from the
+    source data into the destination table.
+
+    When using this function, make sure to provide the right destination table.
+
+    :param db_connector: PostgreSQL database connector
+    :param df: Dataframe containing the source data
+    :param schema: database schema in which the destination table is stored, generally is 'public'
+    :param table_name: name of the destination table
+    """
+    db_cursor = db_connector.cursor()
+    try:
+        db_cursor.execute(
+            sql.SQL("TRUNCATE {table};").format(table=sql.Identifier(schema, table_name))
+        )
+        table_column_names = _get_column_names_from_table(
+            db_cursor, schema=schema, table_name=table_name
+        )
+        df_column_names = df.columns.to_list()
+        _check_columns(table_column_names, df_column_names)
+        for index, row in df.iterrows():
+            db_cursor.execute(
```

Comment:
It might be interesting to use the `copy_from` method instead of inserting (cf. [docs](https://www.psycopg.org/docs/cursor.html#cursor.copy_from))